ProcessLib/TES/TESLocalAssembler-impl.h

```diff
@@ -0,0 +1,255 @@
+
+        for (unsigned c=0; c<mat.cols(); ++c)
+        {
+            switch (MATRIX_OUTPUT_FORMAT)
+            {
+            case MatOutType::OGS5:
+                std::printf(" %.16e", mat(r, c));
+                break;
+            case MatOutType::PYTHON:
+                if (c!=0) std::printf(",");
+                std::printf(" %23.16g", mat(r, c));
+                break;
+            }
+
+        }
+
+        switch (MATRIX_OUTPUT_FORMAT)
+        {
+        case MatOutType::OGS5:
+            std::printf(" | ");
+            break;
+        case MatOutType::PYTHON:
+            std::printf(" ]");
+            break;
+        }
+    }
+    std::printf("\n");
+}
+
+template<typename Vec>
+void
+ogs5OutVec(const Vec& vec)
+{
+    for (unsigned r=0; r<vec.size(); ++r)
+    {
+        switch (MATRIX_OUTPUT_FORMAT)
+        {
+        case MatOutType::OGS5:
+            if (r!=0) std::printf("\n");
+            std::printf("| %.16e | ", vec[r]);
+            break;
+        case MatOutType::PYTHON:
+            if (r!=0) std::printf(",\n");
+            std::printf("[ %23.16g ]", vec[r]);
+            break;
+        }
+    }
+    std::printf("\n");
+}
+
+} // anonymous namespace
+
+namespace ProcessLib
+{
+
+namespace TES
+{
+
+template <typename ShapeFunction_,
+          typename IntegrationMethod_,
+          typename GlobalMatrix,
+          typename GlobalVector,
+          unsigned GlobalDim>
+TESLocalAssembler<ShapeFunction_,
+    IntegrationMethod_,
+    GlobalMatrix,
+    GlobalVector,
+    GlobalDim>::
+TESLocalAssembler(MeshLib::Element const& e,
+                  std::size_t const /*local_matrix_size*/,
+                  unsigned const integration_order,
+                  AssemblyParams const& asm_params)
+    : _shape_matrices(
+          initShapeMatrices<ShapeFunction, ShapeMatricesType, IntegrationMethod_, GlobalDim>(
+              e, integration_order))
+    , _d(asm_params,
+         // TODO narrowing conversion
+         static_cast<const unsigned>(_shape_matrices.front().N.rows()) /* number of integration points */,
+         GlobalDim)
+    , _local_M(ShapeFunction::NPOINTS * NODAL_DOF, ShapeFunction::NPOINTS * NODAL_DOF)
+    , _local_K(ShapeFunction::NPOINTS * NODAL_DOF, ShapeFunction::NPOINTS * NODAL_DOF)
+    , _local_b(ShapeFunction::NPOINTS * NODAL_DOF)
+    , _integration_order(integration_order)
+{
+}
+
+template <typename ShapeFunction_,
+          typename IntegrationMethod_,
+          typename GlobalMatrix,
+          typename GlobalVector,
+          unsigned GlobalDim>
+void
+TESLocalAssembler<ShapeFunction_,
+    IntegrationMethod_,
+    GlobalMatrix,
+    GlobalVector,
+    GlobalDim>::
+assemble(const double /*t*/, std::vector<double> const& local_x)
+{
+    _local_M.setZero();
+    _local_K.setZero();
+    _local_b.setZero();
+
+    IntegrationMethod_ integration_method(_integration_order);
+    unsigned const n_integration_points = integration_method.getNPoints();
+
+    _d.preEachAssemble();
```

Comment:
But here is not the right place to do that. I'd like to do documentation in the frame of a general cleanup later on.